app_legacy/Helpers/database/message.php

```diff
@@ -227,6 +227,10 @@ function DeleteMessage(string $user, int $messageID): bool
         return false;
     }
 
+    // If an unread message is deleted by the user, the total unread
+    // messages count displayed on the site will not be updated correctly.
+    markMessageAsRead($user, $messageID);
```

Comment:
This marks the message as read, then calls `UpdateCachedUnreadTotals`. Couldn't you just call that after deleting the message?

Reply:
I think maybe I was going crazy this morning. Invoking `UpdateCachedUnreadTotals()` seems to work. Resolved in https://github.com/RetroAchievements/RAWeb/pull/1489/commits/0213a998097e645dbd5d798ccbf9d45242c689db.